### ISSUE TITLE
"Netflix error.Check the log file"

### ISSUE THREAD

**Issue by sstavross72** [asciidisco/plugin.video.netflix#315]
*I'm submitting a ...*
  - [x] bug report
  - [ ] feature request
  - [ ] support request

### Prerequisites

* [ ] Are you running the latest version?
Yes

### Description
Netflix doesn't start, i am on kodi 18 libreelec latest test build
[Description of the bug or feature]

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:** [What you expected to happen]

**Actual behavior:** [What actually happened]

### Installation

* [x] I installed the plugin via zip from the Releases page
* [ ] I´m using the Netflix Repo
* [ ] I´m using a different source (Please tell which)

### Operating System

* [ ] Linux (x86/x64)
* [ ] OSX (x86/x64)
* [ ] Windows (x86/x64)
* [x] Linux (ARM) Pi3
* [ ] Android

#### Additional informatin on the environment

[Descripe your environment a bit more detailed, are you using LibreElec f.e.]

## Debug log
https://paste.ubuntu.com/p/DVCPCVy4tk/
[Please include a link to your debug log (use something like [http://sprunge.us/](http://sprunge.us/)) or similar, please do not paste]

## Other information

[e.g. detailed explanation, related issues, suggestions how to fix, links for us to have context, etc.]

### Screenshots

[Please add a screenshot if that helps understanding your problem]

[You can erase any parts of this template not applicable to your Issue.]

**Comment by atrias** [comment]
I have the same error on rbp3 with [osmc kodi 18 testing build 17.8-194](https://discourse.osmc.tv/t/testing-kodi-18-leia-builds-for-raspberry-pi/20631)

**Comment by Onslaught-01** [comment]
I have the same error on a linux mint build. I have given up on this ever working as I have never been able to get it working on any build however I really really want it to work.  I have tried installing every dependency I can find that might remotely have anything to do with kodi or the netflix plugin but nothing works. I reinstalled many times, deleted my .kodi directory and reinstalled the addon etc to no avail. If there is anything I can do or you need some error log please message me with the information as to where the file is located and what it is called or any information and I will be glad to help. I have tried getting this app to work on kodi 16,17, and 18 and none have worked. It may be an issue with linux mint specifically i don't know.

**Comment by caphm** [comment]
**TL;DR:** Non-ascii characters in your profile names are the problem (e.g. greek, kyrillic, arabic, ...). This is an unfixed issue within the used function of Python itself. For a quick workaround, rename your profiles to something that does not contain any fancy characters (plain ascii).

**Background:**
It's an issue with your profile names containing non-ascii characters, specifically greek ones (see http://www.utf8-chartable.de/unicode-utf8-table.pl?start=896&number=128&names=-&utf8=0x for the characters starting with 0xce).

This throws an error, because the addon is using `HTMLParser.unescape()` to convert HTML entities to their human readable glyphs. Unfortunately, it is a known issue that this function does not support non-ascii characters (see https://bugs.python.org/issue3932). As far as I can see from a quick google search, there doesn't really exist an elegant und uniformly compatible way to unescape HTML entities in Python 2.7.

I'll look into it and do a PR if I can solve this. Until then, just use the workaround described above.

**Comment by sstavross72** [comment]
Dear caphm,

Thank you very much for your detailed answer!! 

The problem is that I haven't any greek characters in my profile name! I don't even have created a new profile, I use the default one(Master user). I changed the skin's language to English but I have the same problem/error....  The only thing I can imagine now is that perchaps the netflix addons can "see'' my region and tries to load the "greek" version of Netfilx. Maybe characters there create the problem?

**Comment by caphm** [comment]
I don't properly know the internals of how the Netflix API is accessed, but I recall that it parses some info from the usual web page. If that is in Greek, this could be the cause. Try changing the language of your web UI to English.

**Comment by sstavross72** [comment]
Well my friend, success!!! The problem was that I had 2 profiles in Netflix, the second one had greek characters! I edited the second profile, put only Latin characters and that was it!! Netfix now WORKS on my Pi3 running OSMC (kodi 18)!!! So I can have a greek (or other languges) interface but not greek characters in my netflix profiles.
But (as usual) some other problems poped up! Movies don't have any greek subtitles, only english or ...Turkish! At the web interface of netflix (in a PC using Firefox) I have the option to select greek subtitles. Another thing is that the second profile I have is for kid, a default profile of netflix. So at pi3 in this kid's profile I can select greek subtitles for kid's movies, I have also the option to select greek audio! So , something is going on with the Netflix API...

Anyway , thank you very much for your help!! Wish you best of luck!!

**Comment by caphm** [comment]
Glad I could help. The subtitles are a known issue, see #95.

**Comment by atrias** [comment]
In my case the error was appearing without having any non-english characters on any of the profiles.
Anyway I solved the error by re-installing the addon in kodi. Works fine after that!

**Comment by Hallonpaj1** [comment]
Hello, i got the same problem. Running LibreElec 9.0 and Kodi 18.0, and when i start a movie on Netflix Addon the buffering symbol comes on 2 times then nothing happens.

**Comment by derhaegr** [comment]
Same here, LibreElec 9.0. was working fine until now
